### ISSUE TITLE
Automatically search through ForeignKey fields?

### ISSUE THREAD

**Issue by kevinrenskers** [ahmedaljawahiry/django-admin-site-search#20]
I have many models that have a ForeignKey field to the User model. When I use django-admin-site-search to search for "someuser@example.com" by default I only get that user as a result, not also all the objects that belong to this user. 

I can extend the filter like so:

```python

class MyAdminSite(AdminSiteSearchView, admin.AdminSite):
    site_header = "Sound Radix site admin"

    def filter_field(self, query, field):
        if isinstance(field, ForeignKey) and field.related_model is User:
            return Q(**{f"{field.name}__email__icontains": query})

        return super().filter_field(query, field)
```

And then I see many more results, as I'd expect. But it's kind of annoying to have to add these filters by hand for all possible related models (obviously I have ForeignKey fields to other models too). Can this be automated somehow?

**Comment by ahmedaljawahiry** [comment]
Yeah - I think having this as an opt-in via the admin's search fields, as per https://github.com/ahmedaljawahiry/django-admin-site-search/issues/22, is the way to go. Adding it on top of the current implementation would make performance even more problematic.

**Comment by ahmedaljawahiry** [comment]
Available in 1.0.0 👉 https://github.com/ahmedaljawahiry/django-admin-site-search/releases/tag/1.0.0
